client/proposal-scripting-system.md

```diff
@@ -0,0 +1,168 @@
+# Scripting System RFC
+Scripting is a part of the Assets System and is integral in extending the client's capabilities. Scripts are written in Typescript and is compiled with the Typescript Compiler at runtime and run under the V8 Javascript engine.
+
+## Table of Contents
+1. Intents
+2. Commands
+    - Defining Commands
+    - Invoking Commands
+3. Configuration
+    - Getting and Setting Values
+    - Listening to Changes
+4. Entities
+5. Tracking
+6. Notifications
+    - Simple Notifications
+    - Progress Notifications
+    - Indeterminate Notifications
+7. Localization
+
+## Intents
+Intents are integral in making the scripting system secure as the developer must explicitly state featuresets they may request access from the client. Although not fool-proof, transparency to the end-user is enough to know what the extension is capable of doing with the client.
+
+The developer may define intents it may need from the asset's metadata file. The possible values are:
+
+|Intent Name|Description|
+|-----------|-----------|
+|`data`|The script may read from or write to disk in the its given directory.|
+|`network`|The script may download or upload resources online.|
+|`entities`|The script may create and destroy its own entities.|
+|`scene`|The script may access the scene's root and access to other entities.|
+|`tracking`|The script may access tracking information.|
+|`notifications`|The script may create notifications.|
+|`system`|The script may access the client. This is only for internal use.|
+
+## Commands
+Commands are central to how interaction is possible between the script to the client and the script to other scripts.
+
+### Defining Commands
+In defining a command, it must be written in a Typescript module and must have a decorator defined to a function. By default, commands are accessible via the command palette.
+
+*Example:*
+```ts
+@command
+function sayHello(): void {
+    // TODO: Write code that says hello to the user.
+}
+```
+
+### Invoking Commands
+Other scripts can also invoke commands. The string format of a command's ID is:
+
+`<asset-name>:<module-name>.<function-name>`
```

Comment:
We can also infer that this could be a method name too unless you plan only to support functions only